zstash/globus.py

```diff
@@ -31,16 +34,61 @@
     r"perlmutter.*\.nersc\.gov": "6bdc7956-fc0f-4ad2-989c-7aa5ee643a79",
 }
 
-remote_endpoint = None
-local_endpoint = None
-transfer_client: TransferClient = None
-transfer_data: TransferData = None
-task_id = None
-archive_directory_listing: IterableTransferResponse = None
+# Last updated 2025-04-08
+ENDPOINT_TO_NAME_MAP = {
+    "08925f04-569f-11e7-bef8-22000b9a448b": "Invalid, presumably Theta",
+    "15288284-7006-4041-ba1a-6b52501e49f1": "LCRC Improv DTN",
+    "68fbd2fa-83d7-11e9-8e63-029d279f7e24": "pic#compty-dtn",
+    "6bdc7956-fc0f-4ad2-989c-7aa5ee643a79": "NERSC Perlmutter",
+    "6c54cade-bde5-45c1-bdea-f4bd71dba2cc": "Globus Tutorial Collection 1",  # The Unit test endpoint
+    "9cd89cfd-6d04-11e5-ba46-22000b92c6ec": "NERSC HPSS",
+    "de463ec4-6d04-11e5-ba46-22000b92c6ec": "Invalid, presumably ALCF HPSS",
+}
+
+# Helper functions ############################################################
+
+
+def ep_to_name(endpoint_id: str) -> str:
+    if endpoint_id in ENDPOINT_TO_NAME_MAP:
+        return ENDPOINT_TO_NAME_MAP[endpoint_id]
+    else:
+        return endpoint_id  # Just use the endpoint_id itself
+
+
+def log_current_endpoints(globus_info: GlobusInfo):
+    if globus_info.local_endpoint:
+        local = ep_to_name(globus_info.local_endpoint)
+    else:
+        local = "undefined"
+    logger.debug(f"local endpoint={local}")
+    if globus_info.remote_endpoint:
+        remote = ep_to_name(globus_info.remote_endpoint)
+    else:
+        remote = "undefined"
+    logger.debug(f"remote endpoint={remote}")
+
+
+def set_clients(globus_info: GlobusInfo):
+    native_client = NativeClient(
+        client_id=ZSTASH_CLIENT_ID,
+        app_name="Zstash",
+        default_scopes="openid urn:globus:auth:scope:transfer.api.globus.org:all",
+    )
+    log_current_endpoints(globus_info)
+    logger.debug(
+        "set_clients. Calling login, which may print 'Please Paste your Auth Code Below:'"
+    )
+    native_client.login(no_local_server=True, refresh_tokens=True)
+    transfer_authorizer = native_client.get_authorizers().get("transfer.api.globus.org")
+    globus_info.transfer_client = TransferClient(authorizer=transfer_authorizer)
```

Comment:
I initially tried to change out this logic with that in #349, but it didn't seem to change terribly much. 

It actually also complicated things because it uses a different object type than the `native_client` instantiation that's now found in `check_consents()`